core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.PropertyKey;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Task for validating security configurations.
+ */
+public class SecureHdfsValidationTask implements ValidationTask {
+  private static final String LINE_SEPARATOR = System.getProperty("line.separator").toString();
+  private static final String ALLUXIO_BIN = "./bin/alluxio";
+  private static final String GETCONF_CMD = "getConf";
+  private static final Pattern PRINCIPAL_PATTERN =
+      Pattern.compile("(?<primary>[\\w][\\w-]*\\$?)(/(?<instance>[\\w]+))?(@(?<realm>[\\w]+))?");
+
+  private final String mProcess;
+
+  /**
+   * Constructor of {@link SecureHdfsValidationTask}.
+   *
+   * @param process type of the process on behalf of which this validation task is run
+   */
+  public SecureHdfsValidationTask(String process) {
+    mProcess = process.toLowerCase();
+  }
+
+  @Override
+  public boolean validate() {
+    // Check whether can login with specified principal and keytab
+    String principal = "";
+    String keytab = "";
+    String primary;
+    String instance;
+    String realm;
+    if (mProcess.equals("master")) {
+      principal = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
```

Comment:
Please use `Configuration.get(PropertyKey. MASTER_PRINCIPAL)` to get property values instead.

Reply:
Here I need the <b>name</b> of the property, rather than its <b>value</b>.

Reply:
I think I do what you were saying. Will update.